File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/MemberRepository.java
package com.raxim.myscoutee.profile.repository.mongo;

import com.raxim.myscoutee.profile.data.document.mongo.Member;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;
import java.util.UUID;

@RepositoryRestResource(collectionResourceRel = "members", path = "members")
public interface MemberRepository extends MongoRepository<Member, UUID>, QuerydslPredicateExecutor<Member> {
    // Add any additional methods or custom queries if needed
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/MemberRepository.kt
package com.raxim.myscoutee.profile.repository.mongo

import com.raxim.myscoutee.profile.data.document.mongo.Member
import com.raxim.myscoutee.profile.data.document.mongo.Path
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import java.util.*

@RepositoryRestResource(
    collectionResourceRel = "members",
    path = "members"
)
interface MemberRepository : MongoRepository<Member, UUID>,
    QuerydslPredicateExecutor<Member>